### DISCUSSION TITLE
Generowanie szczegółów zwrotu konsumenckiego

### DISCUSSION THREAD

**Discussion by jezewskiBNetrox** [allegro/allegro-api#10131] [Suggestions - rejected]
Dzień dobry,

chciałbym zgłosić potrzebę dodania do zasobów API nowego zasobu, który by generował dokument szczegółów zwrotu w formie PDF. 
Miałby to być odpowiednik takiej funkcjonalności na portalu, która jest w zakładce Zwroty (trzy kropki na zwrocie -> GENERUJ SZCZEGÓŁY ZWROTU (PDF)). Obecnie API nie udostępnia takiej możliwości a zasoby dotyczące zwrotów konsumenckich są jeszcze w becie, więc przypuszczam że jest jeszcze możliwość tam coś takiego wcisnąć. :)

Myślę, że byłoby to akceptowalne gdyby to funkcjonowało w podobie to API używanego w Sales Center.
GET https://api.{environment}/order/customer-returns/{customerReturnId}/form-document

Byłoby to bardzo przydatne dla jednego z moich klientów.
Pozdrawiam.

**Comment by AureliuszBrussy** [comment]
Dziękuję za sugestię. Jeśli zdecydujemy się na wprowadzenie takiej informacji, będziemy komunikować to na naszej stronie i repozytorium.

**Reply by slawek124** [reply]
Podpinam się do tej prośby.

**Comment by slawek124** [comment]
@jezewskiBNetrox do czasu zanim allegro wprowadzi taką metodę, można to rozwiązać na zasadzie funkcji otwierającej odpowiednio przygotowany link do formularza w przeglądarce, pobraniu i zapisaniu pliku. Sposób może archaiczny ale działa. Albo samemu na podstawie [GET /order/customer-returns/{customerReturnId}](https://developer.allegro.pl/documentation/#operation/getCustomerReturnById) wygenerować taki plik pdf, ale to już wymaga znacznie więcej pracy, aby taki dokument był przybliżony do tego z allegro.

**Reply by jezewskiBNetrox** [reply]
> do czasu zanim allegro wprowadzi taką metodę, można to rozwiązać na zasadzie funkcji otwierającej odpowiednio przygotowany link do formularza w przeglądarce, pobraniu i zapisaniu pliku.

@slawek124 Trzeba się zalogować i trzymać sesję, żeby to zadziałało. Niestety Allegro nie lubi botów i automatyzacji na portalu, przez co jest z tym masa problemów. Z braku laku robiłem w taki sposób ale zaczęły wchodzić blokady IP za podejrzane zachowanie. Trzeba niestety też sporo pracy, żeby to dopracować i naprawiać.

Niestety trzeba generować własne PDFy. W .NET jest bardzo spoko opcja od rodaka: https://github.com/QuestPDF/QuestPDF
Biblioteka pozwala generować PDFy i można odtworzyć taki formularz w dosyć prosty sposób. Wymaga to oczywiście trochę pracy ale nadal to lepsze niż godziny ślęczenia nad botem, który by zaciągał ten pliczek.

**Reply by slawek124** [reply]
Dzięki za poradę z QuestPDF, natomiast jak dotąd nie spotkałem się z blokadą IP. Być może jest to związane z czasem pomiędzy pobraniami, takiego formularza.

**Comment by Lukasz-Zurek** [comment]
Zaktualizowaliśmy status sugestii jako "odrzucona", ponieważ nie planujemy sugerowanej zmiany.
